### ISSUE TITLE
Fails to initiate SQLite on first launch

### ISSUE THREAD

**Issue by filcuk** [leepeuker/movary#233]
Latest version won't start, database not created:
```
PHP Fatal error:  Uncaught Exception: Unable to open database: unable to open database file in /app/vendor/doctrine/dbal/src/Driver/SQLite3/Driver.php:35
```
Full log: [movary_log.txt](https://github.com/leepeuker/movary/files/10418318/_movary_logs.2.txt)

My compose:
```yaml
  movary:
    image: leepeuker/movary:latest
    container_name: movary
    restart: unless-stopped
    ports:
      - "80:80"
    environment:
      TIMEZONE: $TZ
      ENV: production
      DATABASE_MODE: "sqlite"
      TMDB_API_KEY: ${TMDB_API_KEY:?}
      TMDB_ENABLE_IMAGE_CACHING: 1  # Enable to prevent rate limit issues
      LOG_LEVEL: warning
      LOG_ENABLE_STACKTRACE: 0
      LOG_ENABLE_FILE_LOGGING: 1  # Log directory is at `storage/logs`
    volumes:
      - $DOCKERDIR/appdata/movary:/app/storage
```

I've tried adding `user: ...` and have ensured the folder is owned by the docker user, but no change.

**Comment by leepeuker** [comment]
Hey, sorry for your issue. The easiest way to solve this would usually be to use a docker volume to persist the storage data instead of mounting the storage directory to the host. If you want to mount check out [this](https://github.com/leepeuker/movary/issues/229#issuecomment-1381845326) comment. In short: Create the `app/public` directory in your storage mount if it does not already exist and **recreate** the container.

**Comment by filcuk** [comment]
Thanks for the quick response!

I would mention that my setup should tick all the boxes in your linked comment, i.e.:
> mounting storage directly to the host works when the new required directory `<storagepath>/app/public` already exist on the host. Create this directory manually and recreate the container (important, e.g. with `--force-recreate flag`). You may have to set user/group id of the container via e.g. `--user 1000:1000` to match the host storage directory owner.
- mounted dir already exists
- `user: $PUID:$PGID` for the dir owner (docker user) was added to the service (folder was `chown`ed too, just in case)
- container was recreated on each attempt
---
Regardless, as you recommended, using docker volume solves that issue:
```yaml
services:
  movary:
    ...
    volumes:
      - movary:/app/storage
    ...
volumes:
  movary:
```

However, now when I run the command to initialise (`docker exec movary php bin/console.php database:migration:migrate`), it reports that the database is read-only:
```
InvalidArgumentException: There was a problem creating the schema table: SQLSTATE[HY000]: General error: 8 attempt to write a readonly database in /app/vendor/robmorgan/phinx/src/Phinx/Db/Adapter/AbstractAdapter.php:318
```
I've tried explicitly stating `movary:/app/storage:rw`, but that didn't help either.

Could the database be `ro` because movary is trying to run a job on the database that hasn't been initialised yet?
Please see this log: [movary_log_2.txt](https://github.com/leepeuker/movary/files/10418611/_movary_logs.3.txt)
```
movary.ERROR: Could not process job. {"exception":"[object] (Doctrine\\DBAL\\Exception\\TableNotFoundException(code: 0): An exception occurred while executing a query: Unable to prepare statement: 1, no such table: job_queue at /app/vendor/doctrine/dbal/src/Driver/API/SQLite/ExceptionConverter.php:52)\n[previous exception] [object] (Doctrine\\DBAL\\Driver\\SQLite3\\Exception(code: 0): Unable to prepare statement: 1, no such table: job_queue at /app/vendor/doctrine/dbal/src/Driver/SQLite3/Exception.php:16)\n[previous exception] [object] (Exception(code: 0): Unable to prepare statement: 1, no such table: job_queue at /app/vendor/doctrine/dbal/src/Driver/SQLite3/Connection.php:25)"} []
```

**Comment by filcuk** [comment]
I've got it running now.
I've had to remove the previously added `user: $PUID:$PGID`.

Just for reference, my final compose looks like this:
```yaml
services:
  movary:
    image: leepeuker/movary:latest
    container_name: movary
    restart: "no"
    ports:
      - "8712:80"
    environment:
      TIMEZONE: $TZ
      ENV: production
      DATABASE_MODE: "sqlite"
      TMDB_API_KEY: ${TMDB_API_KEY:?}
      TMDB_ENABLE_IMAGE_CACHING: 1  # Enable to prevent rate limit issues
      LOG_LEVEL: warning
      LOG_ENABLE_STACKTRACE: 0
      LOG_ENABLE_FILE_LOGGING: 1  # Log directory is at `storage/logs`
      MIN_RUNTIME_IN_SECONDS_FOR_JOB_PROCESSING: 15 # Min time between jobs
    volumes:
      - movary-vol:/app/storage
volumes:
  movary-vol:
    external: true
```

Thanks for your help!

**Comment by filcuk** [comment]
Actually... I was getting logged out in the web UI on every action, so I restarted the container, and I've encountered the issue #204, even after multiple restarts.

The solution is to re-add `user: $PUID:$PGID` after the first restart.
This does allow the service to start up, but it also still causes the `readonly database` issue mentioned above, which results in login/auth not working.

Very odd, and I'm out of ideas.

**Comment by leepeuker** [comment]
So I finally got time to test your docker-compose file @filcuk 

It worked for me like you posted it. Please try the following steps:
```
# Kill current container
docker compose down

# Recreate volume to clean the slate
docker volume remove movary-vol
docker volume create movary-vol

# Force container recreation
docker compose up -d --build

# Execute database migrations
docker exec movary php bin/console.php database:migration:migrate
```
Does this work for you?

**Comment by leepeuker** [comment]
FYI I have tried the docker-compose file again on the same machine (following the steps I mentioned in my last comment) but with a different user (to test what happens with a different UID) and it worked without an issue as before.

**Comment by filcuk** [comment]
Apologies for taking so long to respond.
Updated to the `:latest` tag today, but no changes so far as I can tell.
I've tried running with MySQL today, but that appears to make no difference.

> Which OS are you using?  

I'm on Synology DSM 7.1 (supposedly a Debian fork)
Docker 20.10.3
docker-compose 1.28.5

---

### To recap / step-by-step
1. Remove containers, volumes
2. Start containers
3. Run migrations `php bin/console.php database:migration:migrate`
4. Register initial user
5. Get redirected to the dashboard (all ok so far)

**A: On localhost:**
1. Any action gets me logged out
2. Log back in
3. Repeat 1.

**B: Through a reverse proxy:**
1. Log a movie -> success
2. Try to navigate anywhere
3. Stuck on loading / server stops responding
4. Restart container
5. Container exits: `error: failed switching to "root": operation not permitted`

---

> Could you try to reproduce this on another host?

Let me get back to you with more details.

**Comment by leepeuker** [comment]
Are you running other containers which write the the filesystem on your Synology NAS? I checked google and there seems to be some sepcial behavior on DSM (e.g. see [here](https://www.reddit.com/r/synology/comments/k10bh1/docker_portainer_and_volume_locations/grdeppz?utm_medium=android_app&utm_source=share&context=3)) in regards to docker, maybe the issue is rooted there.

**Comment by filcuk** [comment]
**Tested the same compose on Windows and it runs flawlessly.**

I am running around 80 other containers.
Everything is done through compose files and SSH.
I read the post, but I think it only clarifies some terminology op was confused about.

Clearly there's some issue with my environment.
I would like to get it running, because it looks superb, especially with the 'scrobblers' for both Plex & Jellyfin!
I'm going to close the issue, but if I find out what was causing it (hopefully not some dumb mistake on my part), I'll post it here.

Thanks again for all your help and suggestions!

**Comment by filcuk** [comment]
I don't know whether this is useful at all, but...
I've just manually copied the contents of the movary container `/app/storage` from the instance running successfully on my desktop to the NAS, then spun the container up.

I can now add movies, browse freely, database does not lock up anymore.

---

I still can't restart/stop the container however.
As soon as I do, I get `error: failed switching to "root": operation not permitted` on next launch, as mentioned previously.

I have tested `docker restart movary` on my Win instance, and it results in the same error.

I've found similar behaviour mentioned here:
https://github.com/webdevops/Dockerfile/issues/366#issuecomment-1191922337
